src/app/clusters/bindings/PendingNotificationMap.cpp

```diff
@@ -31,30 +31,26 @@ CHIP_ERROR PendingNotificationMap::FindLRUConnectPeer(FabricIndex * fabric, Node
     // First, set up a way to easily track which entries correspond to the same peer.
     uint8_t bindingWithSamePeer[EMBER_BINDING_TABLE_SIZE];
 
-    for (uint8_t i = 0; i < EMBER_BINDING_TABLE_SIZE; i++)
+    for (auto iter = BindingTable::GetInstance().begin(); iter != BindingTable::GetInstance().end(); ++iter)
     {
-        EmberBindingTableEntry entry;
-        emberGetBinding(i, &entry);
-        if (entry.type != EMBER_UNICAST_BINDING)
+        if (iter->type != EMBER_UNICAST_BINDING)
         {
             continue;
         }
         bool foundSamePeer = false;
-        for (uint8_t j = 0; j < i; j++)
+        for (auto checkIter = BindingTable::GetInstance().begin(); checkIter != BindingTable::GetInstance().end(); ++checkIter)
```

Comment:
```suggestion
        for (auto checkIter = BindingTable::GetInstance().begin(); checkIter != iter; ++checkIter)
```
Though that said, if we do just go to the end, it's going to run into `iter` and end up doing `bindingWithSamePeer[iter.GetIndex()] = checkIter.GetIndex();` which is exactly what we sould have done anyway in the `!foundSamePeer` case.  So maybe we should just leave this condition as-is, but take out the `foundSamePeer` bits?  But one or the other; we shouldn't have unreachable code.